abdm_integrator/hiu/views/health_information.py

```diff
@@ -0,0 +1,295 @@
+        in response.
+        """
+        request_data = request.query_params
+        RequestHealthInformationSerializer(data=request_data).is_valid(raise_exception=True)
+
+        artefact = get_object_or_404(ConsentArtefact, artefact_id=request_data['artefact_id'],
+                                     consent_request__user=request.user)
+        self.validate_artefact_expiry(artefact)
+
+        current_url = request.build_absolute_uri(reverse('request_health_information'))
+        health_info_url = request.build_absolute_uri(reverse('receive_health_information'))
+        # For multiple pages, client is expected make a new request for subsequent pages with parameters
+        # 'transaction_id' and 'page' as returned from the previous request
+        if request_data.get('transaction_id') and request_data.get('page'):
+            page_number = request_data['page']
+            gateway_request_id = get_object_or_404(
+                HealthInformationRequest, transaction_id=request_data['transaction_id']).gateway_request_id
+        else:
+            page_number = 1
+            hiu_crypto = ABDMCrypto()
+            gateway_request_id = self.gateway_health_information_cm_request(artefact, health_info_url,
+                                                                            hiu_crypto.transfer_material)
+            self.save_health_info_request(request.user, artefact, gateway_request_id,
+                                          hiu_crypto.key_material.as_dict())
+
+        cache_key = f'{gateway_request_id}_{page_number}'
+        health_response_data = poll_for_data_in_cache(cache_key, interval=4)
+
+        self.handle_for_error(health_response_data)
+        if app_settings.HIU_PARSE_FHIR_BUNDLE:
+            health_response_data['entries'] = self.parse_fhir_bundle_for_ui(health_response_data['entries'])
+        response_data = self.generate_response_data(health_response_data, current_url,
+                                                    artefact.artefact_id)
+        return Response(status=HTTP_200_OK, data=response_data)
+
+    def validate_artefact_expiry(self, artefact):
+        if artefact.consent_request.expiry_date <= datetime.utcnow():
+            raise CustomError(
+                error_code=HIUError.CODE_CONSENT_EXPIRED,
+                error_message=HIUError.CUSTOM_ERRORS[HIUError.CODE_CONSENT_EXPIRED],
+            )
+
+    def gateway_health_information_cm_request(self, artefact, health_info_url, hiu_transfer_material):
+        payload = ABDMRequestHelper.common_request_data()
+        payload['hiRequest'] = {'consent': {'id': str(artefact.artefact_id)}}
+        payload['hiRequest']['dateRange'] = artefact.details['permission']['dateRange']
+        payload['hiRequest']['dataPushUrl'] = health_info_url
+        payload['hiRequest']['keyMaterial'] = hiu_transfer_material
+        logging.info(f"HIU: Request Health Information {payload}")
```

Comment:
Where do these logs end up. You should consider configuring a logger/handler if you care for these logs or make sure these don't end up in the main hq log.

Reply:
These logs were added for testing purpose and is not intended to be merged. Have gotten rid of those now .
( FYI - I do have a ticket for logging https://dimagi-dev.atlassian.net/browse/INDIV-191 as it would be beneficial to have them for gateway interactions)